### PR TITLE
[KotlinCleanup] DeckOptions

### PR DESCRIPTION
**Pull request by n1snt**
## How Has This Been Tested?
- Tested on a physical device
- Tests

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented on your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by lukstbit — APPROVED** [review]
Thanks

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 1]
```diff
@@ -584,10 +584,8 @@ class DeckOptions :
                 this.pref.getString(key, "")
             }
             // update summary
-            @KotlinCleanup("Remove `val s` and use pref.summary?.toString()")
             if (!this.pref.mSummaries.containsKey(key)) {
-                val s = pref.summary
-                this.pref.mSummaries[key] = if (s != null) pref.summary.toString() else null
+                this.pref.mSummaries[key] = if (pref.summary != null) pref.summary.toString() else null
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:588]: The `if-else` is not needed anymore:

`this.pref.mSummaries[key] = pref.summary?.toString()`

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 2]
```diff
@@ -607,11 +607,10 @@ class DeckOptions :
         Collections.sort(confs, NamedJSONComparator.INSTANCE)
         val confValues = arrayOfNulls<String>(confs.size)
         val confLabels = arrayOfNulls<String>(confs.size)
-        @KotlinCleanup(".forEachIndexed")
-        for (i in confs.indices) {
-            val o = confs[i]
-            confValues[i] = o.getString("id")
-            confLabels[i] = o.getString("name")
+        confs.indices.forEachIndexed { index, _ ->
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:610]: By using `forEachIndexed` you don't have to iterate over the collection indices, you get the index and the element:

```
confs.forEachIndexed { index, deckConfig ->
   confValues[index] = deckConfig.getString("id")
   confLabels[index] = deckConfig.getString("name")
}
```

**Review by lukstbit — CHANGES_REQUESTED** [review]
Left some issues to be addressed.

**Review by lukstbit — APPROVED** [review]

**Review by ShridharGoel — APPROVED** [review]

**Pull request merged**